src/libs/actions/Report.js

```diff
@@ -904,21 +919,27 @@ function addAction(reportID, text, file) {
  * network layer handle the delayed write.
  *
  * @param {Number} reportID
- * @param {Number} sequenceNumber
+ * @param {Number} [sequenceNumber]
  */
 function updateLastReadActionID(reportID, sequenceNumber) {
-    const currentMaxSequenceNumber = reportMaxSequenceNumbers[reportID];
-    if (sequenceNumber < currentMaxSequenceNumber) {
-        return;
-    }
+    // If we are not specifying the sequence number, let's set it to the max one available
+    const lastReadSequenceNumber = sequenceNumber || reportMaxSequenceNumbers[reportID];
 
-    setLocalLastRead(reportID, sequenceNumber);
+    setLocalLastRead(reportID, lastReadSequenceNumber);
+
+    // If we are specifying the last read sequence number, we are manually marking a comment as unread, so
+    // let's place new marker in the after the last read sequence.
+    if (sequenceNumber) {
+        Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
+            newMarkerSequenceNumber: sequenceNumber + 1,
+        });
+    }
 
     // Mark the report as not having any unread items
     API.Report_UpdateLastRead({
         accountID: currentUserAccountID,
         reportID,
-        sequenceNumber,
+        sequenceNumber: lastReadSequenceNumber || 0,
```

Comment:
Why isn't this logic part of where `lastReadSequenceNumber` is defined?

Reply:
+1 but also curious about the situation where this will be `0` since that will effectively mark a report as completely unread after the API changes we're introducing and seems unlikely to be the desired outcome if:

1. We're setting the last read to the max that we have (and not a specific `sequenceNumber`)
2. For some reason we're missing the max? Why/how exactly could that happen?

Reply:
It seemed to be some sort of race condition. I have not been able to consistently reproduce it, but `sometimes`, lastReadSequenceNumber ended up empty and the API complained in the console.

Reply:
>I have not been able to consistently reproduce it, but sometimes, lastReadSequenceNumber ended up empty and the API complained in the console

Should we maybe throw an error instead of marking the report as unread. It seems like this is possibly the total opposite of what the user wants to do?

Reply:
I `think` we can actually get rid of it now. The issue seemed caused by `reportMaxSequenceNumbersNotLoading`, which we no longer use